conda_package_handling/conda_fmt.py

```diff
@@ -106,3 +112,40 @@ def get_pkg_details(in_file):
         with open(in_file, 'rb') as fd:
             outer_sha256 = utils.sha256_checksum(fd)
         return {"size": size, "inner_sha256": inner_sha256, "outer_sha256": outer_sha256}
+
+    @staticmethod
+    def gpg_sign(fn, gpg_wrapper=None, out_folder=os.getcwd()):
+        assert gpg_wrapper
+
+        final_fn = os.path.join(out_folder, os.path.basename(fn))
+        with TemporaryDirectory() as tmp:
+            with utils.tmp_chdir(tmp):
+                with zipfile.ZipFile(fn, 'r', compression=zipfile.ZIP_STORED) as zf:
+                    zf.extractall()
+                    content = utils.ensure_list(zf.namelist())
+                signature_file = 'signature.asc'
+                try:
+                    content.remove(signature_file)
+                except ValueError:
+                    pass
```

Comment:
This is required in case the artifact is already signed and needs to be signed again. Should I warn if signature already exists?